pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -508,6 +527,57 @@ func (c *Cluster) DeployPrometheusRule(name, namespace string) error {
 	return nil
 }
 
+func (c *Cluster) templateToPrometheusRule(name, templateData string) (*monitoringv1.PrometheusRule, error) {
+	var rule monitoringv1.PrometheusRule
+	customPrometheusRule, err := c.getComputeCustomizedPrometheus()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to compute customized prometheus rule")
+	}
+	t, err := loadTemplate(name, templateData, customPrometheusRule)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load prometheus rule template")
+	}
+	err = yaml.Unmarshal(t, &rule)
+	if err != nil {
+		return nil, err
+	}
+	return &rule, nil
+}
+
+func loadTemplate(name, templateData string, p *cephv1.PrometheusRuleCustomized) ([]byte, error) {
+	var writer bytes.Buffer
+	t := template.New(name)
+	t, err := t.Parse(templateData)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse template %v", name)
+	}
+	err = t.Execute(&writer, p)
+	return writer.Bytes(), err
+}
+
+// getComputeCustomizedPrometheus compute PrometheusRuleCustomized by merging the data that was get from env with the default data
+func (c *Cluster) getComputeCustomizedPrometheus() (*cephv1.PrometheusRuleCustomized, error) {
+	var defaultPrometheusRuleVals cephv1.PrometheusRuleCustomized
+	fi, err := os.Open(filepath.Clean(path.Join(monitoringPath, defaultPrometheusRuleFile)))
```

Comment:
Have you tested this in a K8s cluster or just unit tests? The default file needs to be embedded if this is going to work, but maybe i'm missing it.

Reply:
Yes, I tested it on K8s cluster, it works as expected.
The defaults file is just the data the will populate the template of the [PrometheusRule resource file](https://github.com/rook/rook/blob/b21ed7daf72bd4c8ef58737b30714532a648a7e4/pkg/operator/ceph/cluster/mgr/template/prometheusrule.yaml), which is the file that should be embedded as can be seen [here](https://github.com/rook/rook/blob/b21ed7daf72bd4c8ef58737b30714532a648a7e4/pkg/operator/ceph/cluster/mgr/mgr.go#L60-L61)